### DISCUSSION TITLE
Seeking Clarification on Form API Routes in Sunbird Portal

### DISCUSSION THREAD

**Discussion by rajnishdargan** [Sunbird-Ed/Community#751] [General Discussion]
Hello,

In the Sunbird portal I noticed that the form APIs are being used for various configurations:

`{{host}}/action/data/v1/form/read`

I wanted to understand which service these form APIs belong to. For example, I know that the framework APIs belong to the taxonomy service, and I can find the corresponding routes in the taxonomy service repository here:
[Taxonomy Routes (release-6.0.0)](https://github.com/Sunbird-Knowlg/knowledge-platform/blob/release-6.0.0/taxonomy-api/taxonomy-service/conf/routes)

The specific form API endpoints I’m referring to are:

```
/data/v1/form/read
/data/v1/form/update
/data/v1/form/create
/data/v1/form/fetchAll
```

Could anyone help clarify:

Which service these form APIs belong to?
Where I can find the routes file or service repository for these form APIs?

@pallakartheekreddy @HarishGangula @rajeevsathish @vinukumar-vs @vaibhavbhuva @krgauraw @amitpriyadarshi
@coolbung @itsvick @snehal0904 @arati-tekdi @rushi-tekdi

**Comment by HarishGangula** [comment]
@rajnishdargan  It is part of the Sunbird Ed Portal as [plugin](https://github.com/Sunbird-Ed/SunbirdEd-portal/blob/babc59c755e1ee5fd7e16a7047eb4303557c8d6f/src/app/framework.config.js#L45)

**Reply by rajnishdargan** [reply]
Hi @HarishGangula @pallakartheekreddy,

When checking the plugin for the npm package [@project-sunbird/form-service](https://www.npmjs.com/package/@project-sunbird/form-service), it points to the following repository:
https://github.com/project-sunbird/sunbird-ext-framework/tree/master/demo/plugins/FormService

Could you please guide me on how this NPM package/repository can be used to set up the form service?

We are using the Knowlg generic-editor, which relies on the form-service for various Edit, Review, and Publish forms.

Could you also clarify where the database should be created and provide an overview of how the entire system works?
@coolbung @itsvick @snehal0904 @arati-tekdi @rushi-tekdi

**Reply by HarishGangula** [reply]
@rajnishdargan  You can follow https://github.com/project-sunbird/sunbird-ext-framework/tree/master/server for same
